src/pkcs11/pkcs11-spy.c

```diff
@@ -1020,7 +1038,7 @@ C_Encrypt(CK_SESSION_HANDLE hSession, CK_BYTE_PTR pData, CK_ULONG  ulDataLen,
 	spy_dump_ulong_in("hSession", hSession);
 	spy_dump_string_in("pData[ulDataLen]", pData, ulDataLen);
 	rv = po->C_Encrypt(hSession, pData, ulDataLen, pEncryptedData, pulEncryptedDataLen);
-	if (rv == CKR_OK)
+	if (rv == CKR_OK || rv == CKR_BUFFER_TOO_SMALL)
```

Comment:
If I understand correctly, CKR_BUFFER_TOO_SMALL may set the output length, but do you really think that in this case the output buffer should be filled by the module so that we dump it here?

Reply:
if CKR_BUFFER_TOO_SMALL is returned then pEncryptedData may have any random data in it, C_Encrypt will probably not touch it at all.